Provider Crash When Attempting to Read Missing Variable Resource

**Describe the bug**
The provider will crash if a variable managed by the terraform provider is deleted outside of terraform.

**Steps to reproduce**
1. Create a Library variable set
2. Create a variable in the created variable set
3. Apply
4. Manually delete the variable
5. Run Plan (crashes)

**Expected behavior**
Provider should recognize resource is missing and create it.

**Logs and other supporting information**
```go
octopusdeploy_library_variable_set.this: Refreshing state... [id=LibraryVariableSets-1]
octopusdeploy_variable.this: Refreshing state... [id=47b0a08b-9bae-4034-ade4-7cae4ce915d8]

Error: rpc error: code = Unavailable desc = transport is closing


panic: runtime error: invalid memory address or nil pointer dereference
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa812a6]
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16: 
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16: goroutine 99 [running]:
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16: github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy.setVariable(0xf79520, 0xc00084cf60, 0xc0004bec00, 0x0, 0xc00072e9c0, 0x17)
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16:   github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy/schema_variable.go:121 +0x26
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16: github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy.resourceVariableRead(0xf79520, 0xc00084cf60, 0xc0004bec00, 0xcef3c0, 0xc0001fc000, 0xc000352320, 0xc0000cb8f0, 0x40e118)
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16:   github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy/resource_variable.go:58 +0x4a9
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc000365ea0, 0xf794a0, 0xc000836400, 0xc0004bec00, 0xcef3c0, 0xc0001fc000, 0x0, 0x0, 0x0)
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16:   github.com/hashicorp/terraform-plugin-sdk/v2@v2.5.0/helper/schema/resource.go:347 +0x17c
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000365ea0, 0xf794a0, 0xc000836400, 0xc0004c0cb0, 0xcef3c0, 0xc0001fc000, 0xc000406e10, 0x0, 0x0, 0x0)
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16:   github.com/hashicorp/terraform-plugin-sdk/v2@v2.5.0/helper/schema/resource.go:624 +0x1cb
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0005acac0, 0xf794a0, 0xc000836400, 0xc000836440, 0xc000836400, 0xde0a80, 0xe16280)
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16:   github.com/hashicorp/terraform-plugin-sdk/v2@v2.5.0/helper/schema/grpc_provider.go:575 +0x43b
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16: github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ReadResource(0xc00000df00, 0xf794a0, 0xc000836400, 0xc00084cc00, 0xc00000df00, 0xc00059be00, 0xc0005cdba0)
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16:   github.com/hashicorp/terraform-plugin-go@v0.2.1/tfprotov5/server/server.go:298 +0x105
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler(0xe16280, 0xc00000df00, 0xf79560, 0xc00059be00, 0xc00084cba0, 0x0, 0xf79560, 0xc00059be00, 0xc00072c200, 0x100)
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16:   github.com/hashicorp/terraform-plugin-go@v0.2.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x214
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16: google.golang.org/grpc.(*Server).processUnaryRPC(0xc00042a540, 0xf82ba0, 0xc000093980, 0xc0005d8900, 0xc00061a5a0, 0x14711f0, 0x0, 0x0, 0x0)
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16:   google.golang.org/grpc@v1.36.1/server.go:1217 +0x522
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16: google.golang.org/grpc.(*Server).handleStream(0xc00042a540, 0xf82ba0, 0xc000093980, 0xc0005d8900, 0x0)
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16:   google.golang.org/grpc@v1.36.1/server.go:1540 +0xd05
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16: google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0000ae200, 0xc00042a540, 0xf82ba0, 0xc000093980, 0xc0005d8900)
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16:   google.golang.org/grpc@v1.36.1/server.go:878 +0xa5
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16: created by google.golang.org/grpc.(*Server).serveStreams.func1
2021-04-06T20:21:36.781-0400 [DEBUG] plugin.terraform-provider-octopusdeploy_v0.7.16:   google.golang.org/grpc@v1.36.1/server.go:876 +0x1fd
2021/04/06 20:21:36 [TRACE] vertex "octopusdeploy_variable.this": visit complete
2021-04-06T20:21:36.782-0400 [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021/04/06 20:21:36 [TRACE] vertex "octopusdeploy_variable.this": dynamic subgraph encountered errors
2021/04/06 20:21:36 [TRACE] vertex "octopusdeploy_variable.this": visit complete
2021/04/06 20:21:36 [TRACE] vertex "octopusdeploy_variable.this (expand)": dynamic subgraph encountered errors
2021/04/06 20:21:36 [TRACE] vertex "octopusdeploy_variable.this (expand)": visit complete
2021/04/06 20:21:36 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
2021-04-06T20:21:36.782-0400 [DEBUG] plugin: plugin process exited: path=.terraform/providers/registry.terraform.io/octopusdeploylabs/octopusdeploy/0.7.16/linux_amd64/terraform-provider-octopusdeploy_v0.7.16 pid=5931 error="exit status 2"
2021/04/06 20:21:36 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/octopusdeploylabs/octopusdeploy\"] (close)" errored, so skipping
2021/04/06 20:21:36 [TRACE] dag/walk: upstream of "root" errored, so skipping
2021/04/06 20:21:36 [INFO] backend/local: plan operation completed
2021/04/06 20:21:36 [TRACE] statemgr.Filesystem: removing lock metadata file .terraform.tfstate.lock.info
2021/04/06 20:21:36 [TRACE] statemgr.Filesystem: unlocking terraform.tfstate using fcntl flock
2021-04-06T20:21:36.783-0400 [DEBUG] plugin: plugin exited
```

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Environment and versions:**
- OS: Any
- Octopus Server Version: Any
- Terraform Version: 0.14.7
- Octopus Terraform Provider Version: 0.7.0+

**Additional context**
RCA: Behavior is a result of commit c11f9170cc9741c97057af495f5d8bb7b88faf08 due to the GetByID function of the VariableService in the go api not returning a 404 when a variable is missing. Function works as expected if the project the variable is tied to is manually deleted. 

This can be fixed by either adding back the 
```go
	if variable == nil {
		d.SetId("")
		return nil
	}
```
check, or by changing the Not Found behavior of the GetByID function in the go-octopusdeploy api:

https://github.com/OctopusDeploy/go-octopusdeploy/blob/e4c34bb321928368175aee8eb9775858df0ec98d/octopusdeploy/variable_service.go#L80

to
```go
	return nil, APIError{
		StatusCode:   404,
		ErrorMessage: fmt.Sprintf("Variable %s could not be found in project %s.", variableID, projectID),
	}
```

--- a/docs/data-sources/variables.md
+++ b/docs/data-sources/variables.md
@@ -36,8 +36,8 @@ Read-Only:
 - **is_sensitive** (Boolean) Indicates whether or not this resource is considered sensitive and should be kept secret.
 - **key_fingerprint** (String)
 - **name** (String) The name of this resource.
+- **owner_id** (String)
 - **pgp_key** (String, Sensitive)
-- **project_id** (String)
 - **prompt** (Set of Object) (see [below for nested schema](#nestedatt--variables--prompt))
 - **scope** (List of Object) (see [below for nested schema](#nestedatt--variables--scope))
 - **sensitive_value** (String, Sensitive)
@@ -63,15 +63,7 @@ Read-Only:
 - **channels** (List of String)
 - **environments** (List of String)
 - **machines** (List of String)
-- **parent_deployments** (List of String)
-- **private** (List of String)
-- **process_owners** (List of String)
-- **projects** (List of String)
 - **roles** (List of String)
-- **target_roles** (List of String)
 - **tenant_tags** (List of String)
-- **tenants** (List of String)
-- **triggers** (List of String)
-- **users** (List of String)
 
 

--- a/docs/resources/variable.md
+++ b/docs/resources/variable.md
@@ -18,7 +18,7 @@ description: |-
 ### Required
 
 - **name** (String) The name of this resource.
-- **project_id** (String)
+- **owner_id** (String)
 - **type** (String) The type of variable represented by this resource. Valid types are `AmazonWebServicesAccount`, `AzureAccount`, `Certificate`, `Sensitive`, `String`, or `WorkerPool`.
 
 ### Optional
@@ -57,15 +57,7 @@ Optional:
 - **channels** (List of String) A list of channels that are scoped to this variable value.
 - **environments** (List of String) A list of environments that are scoped to this variable value.
 - **machines** (List of String) A list of machines that are scoped to this variable value.
-- **parent_deployments** (List of String) A list of parent deployments that are scoped to this variable value.
-- **private** (List of String)
-- **process_owners** (List of String) A list of process owners that are scoped to this variable value.
-- **projects** (List of String) A list of projects that are scoped to this variable value.
 - **roles** (List of String) A list of roles that are scoped to this variable value.
-- **target_roles** (List of String) A list of target roles that are scoped to this variable value.
 - **tenant_tags** (List of String) A list of tenant tags that are scoped to this variable value.
-- **tenants** (List of String) A list of tenants that are scoped to this variable value.
-- **triggers** (List of String) A list of triggers that are scoped to this variable value.
-- **users** (List of String) A list of users that are scoped to this variable value.
 
 

--- a/go.mod
+++ b/go.mod
@@ -5,14 +5,13 @@ go 1.15
 require (
 	cloud.google.com/go/storage v1.14.0 // indirect
 	github.com/Microsoft/go-winio v0.4.17 // indirect
-	github.com/OctopusDeploy/go-octopusdeploy v1.7.17
+	github.com/OctopusDeploy/go-octopusdeploy v1.7.21
 	github.com/agext/levenshtein v1.2.3 // indirect
 	github.com/anmitsu/go-shlex v0.0.0-20200514113438-38f4b401e2be // indirect
 	github.com/apparentlymart/go-cidr v1.1.0 // indirect
 	github.com/aws/aws-sdk-go v1.38.21 // indirect
 	github.com/fatih/color v1.10.0 // indirect
 	github.com/gliderlabs/ssh v0.3.2 // indirect
-	github.com/go-playground/validator/v10 v10.5.0 // indirect
 	github.com/go-test/deep v1.0.7 // indirect
 	github.com/google/uuid v1.2.0
 	github.com/gruntwork-io/terratest v0.32.23
@@ -26,7 +25,6 @@ require (
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.5.0
 	github.com/hashicorp/yamux v0.0.0-20210316155119-a95892c5f864 // indirect
 	github.com/jhump/protoreflect v1.8.2 // indirect
-	github.com/jinzhu/copier v0.2.9 // indirect
 	github.com/kevinburke/ssh_config v1.1.0 // indirect
 	github.com/klauspost/compress v1.12.1 // indirect
 	github.com/mitchellh/copystructure v1.1.2 // indirect
@@ -36,10 +34,8 @@ require (
 	github.com/oklog/run v1.1.0 // indirect
 	github.com/stretchr/testify v1.7.0
 	github.com/ulikunitz/xz v0.5.10 // indirect
-	golang.org/x/crypto v0.0.0-20210415154028-4f45737414dc // indirect
 	golang.org/x/net v0.0.0-20210415231046-e915ea6b2b7d // indirect
 	golang.org/x/oauth2 v0.0.0-20210413134643-5e61552d6c78 // indirect
-	golang.org/x/sys v0.0.0-20210415045647-66c3f260301c // indirect
 	golang.org/x/term v0.0.0-20210406210042-72f3dc4e9b72 // indirect
 	google.golang.org/api v0.44.0 // indirect
 	google.golang.org/genproto v0.0.0-20210416161957-9910b6c460de // indirect

--- a/octopusdeploy/data_source_variables.go
+++ b/octopusdeploy/data_source_variables.go
@@ -17,7 +17,7 @@ func dataSourceVariable() *schema.Resource {
 }
 
 func dataSourceVariableReadByName(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	projectID := d.Get("project_id")
+	ownerID := d.Get("owner_id")
 	name := d.Get("name")
 	scope := octopusdeploy.VariableScope{}
 
@@ -26,15 +26,15 @@ func dataSourceVariableReadByName(ctx context.Context, d *schema.ResourceData, m
 	}
 
 	client := m.(*octopusdeploy.Client)
-	variables, err := client.Variables.GetByName(projectID.(string), name.(string), scope)
+	variables, err := client.Variables.GetByName(ownerID.(string), name.(string), &scope)
 	if err != nil {
-		return diag.Errorf("error reading variable from project %s with name %s: %s", projectID, name, err.Error())
+		return diag.Errorf("error reading variable with owner ID %s with name %s: %s", ownerID, name, err.Error())
 	}
 	if variables == nil {
 		return nil
 	}
 	if len(variables) > 1 {
-		return diag.Errorf("found %v variables for project %s with name %s, should match exactly 1", len(variables), projectID, name)
+		return diag.Errorf("found %v variables with owner ID %s with name %s, should match exactly 1", len(variables), ownerID, name)
 	}
 
 	d.SetId(variables[0].ID)

--- a/octopusdeploy/resource_variable.go
+++ b/octopusdeploy/resource_variable.go
@@ -27,10 +27,10 @@ func resourceVariableImport(d *schema.ResourceData, m interface{}) ([]*schema.Re
 
 	importStrings := strings.Split(d.Id(), ":")
 	if len(importStrings) != 2 {
-		return nil, fmt.Errorf("octopusdeploy_variable import must be in the form of ProjectID:VariableID (e.g. Projects-62:0906031f-68ba-4a15-afaa-657c1564e07b")
+		return nil, fmt.Errorf("octopusdeploy_variable import must be in the form of OwnerID:VariableID (e.g. Projects-62:0906031f-68ba-4a15-afaa-657c1564e07b")
 	}
 
-	d.Set("project_id", importStrings[0])
+	d.Set("owner_id", importStrings[0])
 	d.SetId(importStrings[1])
 
 	return []*schema.ResourceData{d}, nil
@@ -40,10 +40,12 @@ func resourceVariableRead(ctx context.Context, d *schema.ResourceData, m interfa
 	log.Printf("[INFO] reading variable (%s)", d.Id())
 
 	id := d.Id()
-	projectID := d.Get("project_id").(string)
+	ownerID := d.Get("owner_id").(string)
+
+	log.Printf(`ID: %s OwnerID: %s`, id, ownerID)
 
 	client := m.(*octopusdeploy.Client)
-	variable, err := client.Variables.GetByID(projectID, id)
+	variable, err := client.Variables.GetByID(ownerID, id)
 	if err != nil {
 		apiError := err.(*octopusdeploy.APIError)
 		if apiError.StatusCode == 404 {
@@ -54,7 +56,7 @@ func resourceVariableRead(ctx context.Context, d *schema.ResourceData, m interfa
 		return diag.FromErr(err)
 	}
 
-	if err := setVariable(ctx, d, *variable); err != nil {
+	if err := setVariable(ctx, d, variable); err != nil {
 		return diag.FromErr(err)
 	}
 
@@ -67,20 +69,20 @@ func resourceVariableCreate(ctx context.Context, d *schema.ResourceData, m inter
 		return diag.FromErr(err)
 	}
 
-	projectID := d.Get("project_id").(string)
+	ownerID := d.Get("owner_id").(string)
 	variable := expandVariable(d)
 
 	log.Printf("[INFO] creating variable: %#v", variable)
 
 	client := m.(*octopusdeploy.Client)
-	tfVar, err := client.Variables.AddSingle(projectID, variable)
+	tfVar, err := client.Variables.AddSingle(ownerID, variable)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
 	for _, v := range tfVar.Variables {
 		if v.Name == variable.Name && v.Type == variable.Type && (v.IsSensitive || v.Value == variable.Value) && v.Description == variable.Description && v.IsSensitive == variable.IsSensitive {
-			scopeMatches, _, err := client.Variables.MatchesScope(v.Scope, variable.Scope)
+			scopeMatches, _, err := client.Variables.MatchesScope(v.Scope, &variable.Scope)
 			if err != nil {
 				return diag.FromErr(err)
 			}
@@ -93,7 +95,7 @@ func resourceVariableCreate(ctx context.Context, d *schema.ResourceData, m inter
 	}
 
 	d.SetId("")
-	return diag.Errorf("unable to locate variable in project %s", projectID)
+	return diag.Errorf("unable to locate variable in project %s", ownerID)
 }
 
 func resourceVariableUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
@@ -104,17 +106,17 @@ func resourceVariableUpdate(ctx context.Context, d *schema.ResourceData, m inter
 	}
 
 	tfVar := expandVariable(d)
-	projID := d.Get("project_id").(string)
+	ownerID := d.Get("owner_id").(string)
 
 	client := m.(*octopusdeploy.Client)
-	updatedVars, err := client.Variables.UpdateSingle(projID, tfVar)
+	updatedVars, err := client.Variables.UpdateSingle(ownerID, tfVar)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
 	for _, v := range updatedVars.Variables {
 		if v.Name == tfVar.Name && v.Type == tfVar.Type && (v.IsSensitive || v.Value == tfVar.Value) && v.Description == tfVar.Description && v.IsSensitive == tfVar.IsSensitive {
-			scopeMatches, _, _ := client.Variables.MatchesScope(v.Scope, tfVar.Scope)
+			scopeMatches, _, _ := client.Variables.MatchesScope(v.Scope, &tfVar.Scope)
 			if scopeMatches {
 				if err := setVariable(ctx, d, v); err != nil {
 					return diag.FromErr(err)
@@ -126,16 +128,16 @@ func resourceVariableUpdate(ctx context.Context, d *schema.ResourceData, m inter
 	}
 
 	d.SetId("")
-	return diag.Errorf("unable to locate variable in project %s", projID)
+	return diag.Errorf("unable to locate variable in project %s", ownerID)
 }
 
 func resourceVariableDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	log.Printf("[INFO] deleting variable (%s)", d.Id())
 
-	projID := d.Get("project_id").(string)
+	ownerID := d.Get("owner_id").(string)
 
 	client := m.(*octopusdeploy.Client)
-	_, err := client.Variables.DeleteSingle(projID, d.Id())
+	_, err := client.Variables.DeleteSingle(ownerID, d.Id())
 	if err != nil {
 		return diag.FromErr(err)
 	}

--- a/octopusdeploy/schema_action_template_parameter.go
+++ b/octopusdeploy/schema_action_template_parameter.go
@@ -6,7 +6,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 )
 
-func expandActionTemplateParameter(tfTemplate map[string]interface{}) *octopusdeploy.ActionTemplateParameter {
+func expandActionTemplateParameter(tfTemplate map[string]interface{}) octopusdeploy.ActionTemplateParameter {
 	actionTemplateParameter := octopusdeploy.NewActionTemplateParameter()
 
 	propertyValue := octopusdeploy.NewPropertyValue(tfTemplate["default_value"].(string), false)
@@ -17,15 +17,15 @@ func expandActionTemplateParameter(tfTemplate map[string]interface{}) *octopusde
 	actionTemplateParameter.Label = tfTemplate["label"].(string)
 	actionTemplateParameter.Name = tfTemplate["name"].(string)
 
-	return actionTemplateParameter
+	return *actionTemplateParameter
 }
 
-func expandActionTemplateParameters(actionTemplateParameters []interface{}) []*octopusdeploy.ActionTemplateParameter {
+func expandActionTemplateParameters(actionTemplateParameters []interface{}) []octopusdeploy.ActionTemplateParameter {
 	if len(actionTemplateParameters) == 0 {
 		return nil
 	}
 
-	expandedActionTemplateParameters := []*octopusdeploy.ActionTemplateParameter{}
+	expandedActionTemplateParameters := []octopusdeploy.ActionTemplateParameter{}
 	for _, actionTemplateParameter := range actionTemplateParameters {
 		actionTemplateParameterMap := actionTemplateParameter.(map[string]interface{})
 		expandedActionTemplateParameters = append(expandedActionTemplateParameters, expandActionTemplateParameter(actionTemplateParameterMap))
@@ -33,7 +33,7 @@ func expandActionTemplateParameters(actionTemplateParameters []interface{}) []*o
 	return expandedActionTemplateParameters
 }
 
-func flattenActionTemplateParameters(actionTemplateParameters []*octopusdeploy.ActionTemplateParameter) []interface{} {
+func flattenActionTemplateParameters(actionTemplateParameters []octopusdeploy.ActionTemplateParameter) []interface{} {
 	flattenedActionTemplateParameters := make([]interface{}, 0)
 	for _, actionTemplateParameter := range actionTemplateParameters {
 		a := make(map[string]interface{})

--- a/octopusdeploy/schema_extension_settings_values.go
+++ b/octopusdeploy/schema_extension_settings_values.go
@@ -2,11 +2,11 @@ package octopusdeploy
 
 import "github.com/OctopusDeploy/go-octopusdeploy/octopusdeploy"
 
-func expandExtensionSettingsValues(extensionSettingsValues []interface{}) []*octopusdeploy.ExtensionSettingsValues {
-	expandedExtensionSettingsValues := make([]*octopusdeploy.ExtensionSettingsValues, len(extensionSettingsValues))
+func expandExtensionSettingsValues(extensionSettingsValues []interface{}) []octopusdeploy.ExtensionSettingsValues {
+	expandedExtensionSettingsValues := make([]octopusdeploy.ExtensionSettingsValues, len(extensionSettingsValues))
 	for _, extensionSettingsValue := range extensionSettingsValues {
 		extensionSettingsValueMap := extensionSettingsValue.(map[string]interface{})
-		expandedExtensionSettingsValues = append(expandedExtensionSettingsValues, &octopusdeploy.ExtensionSettingsValues{
+		expandedExtensionSettingsValues = append(expandedExtensionSettingsValues, octopusdeploy.ExtensionSettingsValues{
 			ExtensionID: extensionSettingsValueMap["extension_id"].(string),
 			Values:      extensionSettingsValueMap["values"].([]interface{}),
 		})

--- a/octopusdeploy/schema_project.go
+++ b/octopusdeploy/schema_project.go
@@ -22,7 +22,7 @@ func expandProject(d *schema.ResourceData) *octopusdeploy.Project {
 	}
 
 	if v, ok := d.GetOk("auto_deploy_release_overrides"); ok {
-		project.AutoDeployReleaseOverrides = v.([]*octopusdeploy.AutoDeployReleaseOverride)
+		project.AutoDeployReleaseOverrides = v.([]octopusdeploy.AutoDeployReleaseOverride)
 	}
 
 	if v, ok := d.GetOk("cloned_from_project_id"); ok {

--- a/octopusdeploy/schema_variable.go
+++ b/octopusdeploy/schema_variable.go
@@ -8,7 +8,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-func expandVariable(d *schema.ResourceData) octopusdeploy.Variable {
+func expandVariable(d *schema.ResourceData) *octopusdeploy.Variable {
 	name := d.Get("name").(string)
 
 	variable := octopusdeploy.NewVariable(name)
@@ -56,7 +56,7 @@ func expandVariable(d *schema.ResourceData) octopusdeploy.Variable {
 		}
 	}
 
-	return *variable
+	return variable
 }
 
 func getVariableDataSchema() map[string]*schema.Schema {
@@ -100,7 +100,7 @@ func getVariableSchema() map[string]*schema.Schema {
 			Sensitive: true,
 			Type:      schema.TypeString,
 		},
-		"project_id": {
+		"owner_id": {
 			Required: true,
 			Type:     schema.TypeString,
 		},
@@ -131,7 +131,11 @@ func getVariableSchema() map[string]*schema.Schema {
 	}
 }
 
-func setVariable(ctx context.Context, d *schema.ResourceData, variable octopusdeploy.Variable) error {
+func setVariable(ctx context.Context, d *schema.ResourceData, variable *octopusdeploy.Variable) error {
+	if d == nil || variable == nil {
+		return fmt.Errorf("error setting scope")
+	}
+
 	d.Set("description", variable.Description)
 	d.Set("is_editable", variable.IsEditable)
 	d.Set("is_sensitive", variable.IsSensitive)

--- a/octopusdeploy/schema_variable_scope.go
+++ b/octopusdeploy/schema_variable_scope.go
@@ -18,29 +18,12 @@ func expandVariableScope(flattenedVariableScope interface{}) octopusdeploy.Varia
 	flattenedMap := list[0].(map[string]interface{})
 
 	variableScope := octopusdeploy.VariableScope{
-		Machines:          getSliceFromTerraformTypeList(flattenedMap["machines"]),
-		ParentDeployments: getSliceFromTerraformTypeList(flattenedMap["parent_deployments"]),
-		Private:           getSliceFromTerraformTypeList(flattenedMap["private"]),
-		ProcessOwners:     getSliceFromTerraformTypeList(flattenedMap["process_owners"]),
-		Projects:          getSliceFromTerraformTypeList(flattenedMap["projects"]),
-		Roles:             getSliceFromTerraformTypeList(flattenedMap["roles"]),
-		TargetRoles:       getSliceFromTerraformTypeList(flattenedMap["target_roles"]),
-		Tenants:           getSliceFromTerraformTypeList(flattenedMap["tenants"]),
-		TenantTags:        getSliceFromTerraformTypeList(flattenedMap["tenant_tags"]),
-		Triggers:          getSliceFromTerraformTypeList(flattenedMap["triggers"]),
-		Users:             getSliceFromTerraformTypeList(flattenedMap["users"]),
-	}
-
-	if v, ok := flattenedMap["actions"]; ok {
-		variableScope.Actions = getSliceFromTerraformTypeList(v)
-	}
-
-	if v, ok := flattenedMap["channels"]; ok {
-		variableScope.Channels = getSliceFromTerraformTypeList(v)
-	}
-
-	if v, ok := flattenedMap["environments"]; ok {
-		variableScope.Environments = getSliceFromTerraformTypeList(v)
+		Actions:      getSliceFromTerraformTypeList(flattenedMap["actions"]),
+		Channels:     getSliceFromTerraformTypeList(flattenedMap["channels"]),
+		Environments: getSliceFromTerraformTypeList(flattenedMap["environments"]),
+		Machines:     getSliceFromTerraformTypeList(flattenedMap["machines"]),
+		Roles:        getSliceFromTerraformTypeList(flattenedMap["roles"]),
+		TenantTags:   getSliceFromTerraformTypeList(flattenedMap["tenant_tags"]),
 	}
 
 	return variableScope
@@ -69,44 +52,16 @@ func flattenVariableScope(scope octopusdeploy.VariableScope) []interface{} {
 		flattenedScope["machines"] = scope.Machines
 	}
 
-	if len(scope.ParentDeployments) > 0 {
-		flattenedScope["parent_deployments"] = scope.ParentDeployments
-	}
-
-	if len(scope.ProcessOwners) > 0 {
-		flattenedScope["private"] = scope.ProcessOwners
-	}
-
 	if len(scope.Machines) > 0 {
 		flattenedScope["process_owners"] = scope.Machines
-	}
-
-	if len(scope.Projects) > 0 {
-		flattenedScope["projects"] = scope.Projects
 	}
 
 	if len(scope.Roles) > 0 {
 		flattenedScope["roles"] = scope.Roles
 	}
 
-	if len(scope.TargetRoles) > 0 {
-		flattenedScope["target_roles"] = scope.TargetRoles
-	}
-
-	if len(scope.Tenants) > 0 {
-		flattenedScope["tenants"] = scope.Tenants
-	}
-
 	if len(scope.TenantTags) > 0 {
 		flattenedScope["tenant_tags"] = scope.TenantTags
-	}
-
-	if len(scope.Triggers) > 0 {
-		flattenedScope["triggers"] = scope.Triggers
-	}
-
-	if len(scope.Users) > 0 {
-		flattenedScope["users"] = scope.Users
 	}
 
 	return []interface{}{flattenedScope}
@@ -138,61 +93,14 @@ func getVariableScopeSchema() map[string]*schema.Schema {
 			Optional:    true,
 			Type:        schema.TypeList,
 		},
-		"parent_deployments": {
-			Description: "A list of parent deployments that are scoped to this variable value.",
-			Elem:        &schema.Schema{Type: schema.TypeString},
-			Optional:    true,
-			Type:        schema.TypeList,
-		},
-		"private": {
-			Elem:     &schema.Schema{Type: schema.TypeString},
-			Optional: true,
-			Type:     schema.TypeList,
-		},
-		"process_owners": {
-			Description: "A list of process owners that are scoped to this variable value.",
-			Elem:        &schema.Schema{Type: schema.TypeString},
-			Optional:    true,
-			Type:        schema.TypeList,
-		},
-		"projects": {
-			Description: "A list of projects that are scoped to this variable value.",
-			Elem:        &schema.Schema{Type: schema.TypeString},
-			Optional:    true,
-			Type:        schema.TypeList,
-		},
 		"roles": {
 			Description: "A list of roles that are scoped to this variable value.",
 			Elem:        &schema.Schema{Type: schema.TypeString},
 			Optional:    true,
 			Type:        schema.TypeList,
 		},
-		"target_roles": {
-			Description: "A list of target roles that are scoped to this variable value.",
-			Elem:        &schema.Schema{Type: schema.TypeString},
-			Optional:    true,
-			Type:        schema.TypeList,
-		},
-		"tenants": {
-			Description: "A list of tenants that are scoped to this variable value.",
-			Elem:        &schema.Schema{Type: schema.TypeString},
-			Optional:    true,
-			Type:        schema.TypeList,
-		},
 		"tenant_tags": {
 			Description: "A list of tenant tags that are scoped to this variable value.",
-			Elem:        &schema.Schema{Type: schema.TypeString},
-			Optional:    true,
-			Type:        schema.TypeList,
-		},
-		"triggers": {
-			Description: "A list of triggers that are scoped to this variable value.",
-			Elem:        &schema.Schema{Type: schema.TypeString},
-			Optional:    true,
-			Type:        schema.TypeList,
-		},
-		"users": {
-			Description: "A list of users that are scoped to this variable value.",
 			Elem:        &schema.Schema{Type: schema.TypeString},
 			Optional:    true,
 			Type:        schema.TypeList,

